### ISSUE TITLE
Incorrect Null Reference warnings after Assert.IsNotNull() with 3.0.0

### ISSUE THREAD

**Issue by mus65** [microsoft/testfx#1438]
## Describe the bug

After upgrading to 3.0.0 there are multiple new Null Reference warnings in our tests because the `value` argument in `Assert.IsNotNull()` is missing the `[NotNull]` attribute. It seems like `[NotNull]` was removed with https://github.com/microsoft/testfx/commit/a03fb56c1b5176106f6da8293ee885dba725b85e but it's not obvious to me why and I see no mention of it in the release notes.

This is with .NET 6.0.

## Steps To Reproduce

```
string? test = Environment.ProcessPath;
Assert.IsNotNull(test);
Assert.AreEqual(0, test.Length);
```

## Expected behavior

No Null Reference warning, because IsNotNull makes sure that `test` can't be null in `AreEqual`.

## Actual behavior

Null Reference warning on `test.Length`.

**Comment by Evangelink** [comment]
Just got confirmation that this can be used as post-condition (you can refer to https://learn.microsoft.com/dotnet/csharp/language-reference/attributes/nullable-analysis#postconditions-maybenull-and-notnull).

We need to create a bug fix release for #1437 so we will include this in it too.

**Comment by Evangelink** [comment]
Hey there MSTest v3.0.1 was just released.
